src/dataflow/components/tools/dataflow-tool.tsx

```diff
@@ -55,6 +62,22 @@ export default class DataflowToolComponent extends BaseComponent<IProps, IState>
     this.getContent().setProgram(program);
   }
 
+  private handleSetProgramId = (id: string) => {
+    this.getContent().setProgramId(id);
+  }
+
+  private handleSetProgramStartTime = (time: number) => {
+    this.getContent().setProgramStartTime(time);
+  }
+
+  private handleSetProgramEndTime = (time: number) => {
+    this.getContent().setProgramEndTime(time);
+  }
+
+  private handleSetProgramStartEndTime = (startTime: number, endTime: number) => {
+    this.getContent().setProgramStartEndTime(startTime, endTime);
+  }
+
```

Comment:
Note that a couple of these functions aren't used yet (`handleSetProgramStartTime` and `handleSetProgramEndTime`), but I'm assuming we will need them to handle cases where users are starting and stopping programs (or other cases that we haven't mapped out yet).